core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataUpdateSettingsService.java

```diff
@@ -128,10 +128,10 @@ public void clusterChanged(ClusterChangedEvent event) {
         }
 
         if (nrReplicasChanged.size() > 0) {
-            // update settings and kick of a reroute (implicit) for them to take effect
-            for (final Integer fNumberOfReplicas : nrReplicasChanged.keySet()) {
-                Settings settings = Settings.builder().put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, fNumberOfReplicas).build();
-                final List<Index> indices = nrReplicasChanged.get(fNumberOfReplicas);
+            // update settings and kick off a reroute (implicit) for them to take effect
+            for (final Map.Entry<Integer, List<Index>> nrReplicasChangedWithList : nrReplicasChanged.entrySet()) {
```

Comment:
Was uncertain about the name, I felt this verbosity was needed to explain the contents properly.